### ISSUE TITLE
Typings for Typescript [feature request]

### ISSUE THREAD

**Issue by erichiller** [kriasoft/node-sqlite#5]
Typings for Typescript would be tremendous!

**Comment by tenry92** [comment]
I've created a local definitions file for myself at the moment. It may not conform the guidelines from DefinitelyTyped or whatever, and not 100 % correct / complete yet, but I can work with it for now.

Feel free to update or publish this code.

```typescript
declare module 'sqlite' {
  class Database {
    open(fileName: string, options?: OpenOptions): Promise<Database>;
    close(): Promise<void>;
    run(query: string, ...params: any[]): Promise<RunExecResult>;
    get(query: string, ...params: any[]): Promise<Object>;
    all(query: string, ...params: any[]): Promise<Object[]>;
    exec(query: string): Promise<Database>;
    each(query: string, ...params: any[]): Promise<Statement>;
  }
  
  interface OpenOptions {
    mode?: string;
    verbose?: boolean;
  }
  
  class Statement {
    bind(...params: any[]): Promise<Statement>;
    reset(): Promise<Statement>;
    finalize(): Promise<Statement>;
    run(...params: any[]): Promise<RunExecResult>;
    get(...params: any[]): Promise<Statement>;
    all(...params: any[]): Promise<Statement>;
    each(...params: any[]): Promise<number>;
  }
  
  interface RunExecResult {
    lastID?: number;
    changes?: number;
  }
  
  export = new Database;
}
~~~

**Comment by gms1** [comment]
typescript definitions exist for some time (2014)
see https://github.com/DefinitelyTyped/DefinitelyTyped/commits/master/sqlite3
this can be installed using: 'typings install dt~sqlite3 --global --save' 
or with TypeScript 2.0: 'npm install @types/sqlite3'

btw Tanks for this package!

**Comment by villesinisalo** [comment]
Seems to be resolved by #21
